### ISSUE TITLE
start recording from other apps?

### ISSUE THREAD

**Issue by sudomain** [mafik/echo#18]
Hi I'm looking to start and stop recording at certain times of day. Is there an intent that I can send echo from an automation app to start and stop recording?

**Comment by RichardBronosky** [comment]
I'm also interested. 🪷

**Comment by itanium21** [comment]
+1, would really appreciate possibility to either start/stop recording on some specific times (e.g. I don't need recording while I'm sleeping or while I'm at home) OR have possibility to start/stop recording from some automation apps that would allow me to do the same.

**Comment by m3thm4th** [comment]
The point of this app is that you can leave it running in the background and save only what you need. The only downside of this are battery drain and keeping the mic busy to other apps. For example it's easy to stop it as the last thing you do before going to sleep, and start it as soon as you wake up and pick up the phone. Can you describe a scenario you have in mind?

**Comment by itanium21** [comment]
> ... stop it as the last thing you do before going to sleep, and start it as soon as you wake up and pick up the phone. Can you describe a scenario you have in mind?

Exactly the scenario you just wrote - just do it without requiring manual actions from my side. 
Turning it automatically on when I leave house at time X and off when I arrive at time Y. 
If it would support starting/calling from other apps, e.g. some deep link (or whatever is better), then we could use even more advanced scheduling, e.g. different times during weekends, location based etc.

**Comment by m3thm4th** [comment]
So the recording is automatically discarded because an automatic action has been triggered? Let's say the user goes near home while driving, is the recording disabled and memory lost? Several issues can stem from implementing this.

I understand the appeal of automation, but is it really useful when you interact with your phone so many times a day?

**Comment by sudomain** [comment]
> I understand the appeal of automation, but is it really useful when you interact with your phone so many times a day?

Yes. I don't want to remember to start echo recording every day. I want to configure other apps that can send intents to start/stop it at particular times of day or if I press my volume keys in a particular sequence outside of these times. Neither of these conditions need to be the responsibility of Echo, only the result: the action of starting or stopping Echo's retroactive recording. This issue is only requesting that Echo respond to intents from other apps that will handle the conditions, let "power" users configure those apps, and let these users accept responsibility for edge case scenarios.

**Comment by m3thm4th** [comment]
> I don't want to remember to start echo recording

> I press my volume keys in a particular sequence

ok

**Comment by sudomain** [comment]
> I don't want to remember to start echo recording every day.

>> I don't want to remember to start echo recording

Lol it's almost as if quoting someone out of context can change the meaning of their words. 

> I understand the appeal of automation

I'm glad to hear you now support this feature request haha

**Comment by tobytenax** [comment]
Can this work in Conjunction with an ACR app? I know that it did not work with mine just now, but what I'm trying to ascertain is whether it's a configuration thing or an outright incompatibility

**Comment by dotancohen** [comment]
> I don't need recording while I'm sleeping

I used to think so too. After a single incident, I'm glad that Echo is recording even when I'm not expecting to need it.
